src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -4576,5 +4576,36 @@ public void Test_Issue10321()
 
             result.ExcludingLinterDiagnostics().Should().NotHaveAnyDiagnostics();
         }
+
+        // https://github.com/Azure/bicep/issues/10398
+        [TestMethod]
+        public void Test_Issue10398()
+        {
+            var result = CompilationHelper.Compile(
```

Comment:
Is it worth adding a negative test, to assert that incompatible param & output does raise a diagnostic?